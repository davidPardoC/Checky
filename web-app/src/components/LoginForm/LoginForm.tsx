import AuthServices from "@/services/auth.services";
import { LoginDto } from "@/types/auth";
import {
  Box,
  Button,
  Flex,
  FormControl,
  FormErrorMessage,
  FormLabel,
  Heading,
  Input,
} from "@chakra-ui/react";
import { yupResolver } from "@hookform/resolvers/yup";
import Link from "next/link";
import { useForm } from "react-hook-form";
import * as yup from "yup";

const schema = yup.object({
  email: yup.string().email().required(),
  password: yup.string().required(),
});

export const LoginForm = () => {
  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm({ resolver: yupResolver(schema) });

  const onSubmit = async (data: { [key: string]: any }) => {
    const responseData = await AuthServices.login(data as LoginDto)
    console.log(responseData)
  }

  return (
    <Box boxShadow={"1px 0px 3px"} padding={3} borderRadius={5}>
      <form onSubmit={handleSubmit(onSubmit)}>
        <Heading color={"primary"} textAlign={"center"}>
          Checky
        </Heading>
        <FormControl isRequired isInvalid={!!errors.email}>
          <FormLabel>Email address</FormLabel>
          <Input type="email" {...register("email")} />
          <FormErrorMessage>
            {errors.email?.message as string}
          </FormErrorMessage>
        </FormControl>
        <FormControl isRequired isInvalid={!!errors.password}>
          <FormLabel>Password</FormLabel>
          <Input type="password" {...register("password")} />
          <FormErrorMessage>
            {errors.password?.message as string}
          </FormErrorMessage>
        </FormControl>
        <Flex
          justifyContent={"space-between"}
          alignItems={"center"}
          marginTop={4}
        >
          <Link style={{ textDecoration: "underline" }} href={"/signup"}>
            Do not have an account?
          </Link>
          <Button bg={"primary"} color={"white"} type="submit">
            Iniciar Sesi√≥n
          </Button>
        </Flex>
      </form>
    </Box>
  );
};
