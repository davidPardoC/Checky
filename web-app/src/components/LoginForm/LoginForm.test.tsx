import {render} from "@testing-library/react"
import { LoginForm } from "./LoginForm"

const renderComponent = () => {
    return render(<LoginForm/>)
}

describe("Login Form", ()=>{
    it("should render Login Form", ()=>{
        const {container} = renderComponent()
        expect(container).toMatchSnapshot()
    })
})