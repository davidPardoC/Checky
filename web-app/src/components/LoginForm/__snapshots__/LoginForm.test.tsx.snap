// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form should render Login Form 1`] = `
<div>
  <form>
    <div
      class="chakra-form-control css-0"
      role="group"
    >
      <label
        class="chakra-form__label css-uvho5g"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        Email address
      </label>
      <input
        aria-describedby="field-:r0:-helptext"
        class="chakra-input css-0"
        id="field-:r0:"
        type="email"
      />
      <div
        class="chakra-form__helper-text css-0"
        id="field-:r0:-helptext"
      >
        We'll never share your email.
      </div>
    </div>
  </form>
</div>
`;
